### ISSUE TITLE
Calculating evaluation metrics for only single object on Yolov5 pre-trained model

### ISSUE THREAD

**Issue by parepic** [ultralytics/yolov5#8258]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

My task is to test the accuracy of the pre-trained Yolov5 model in detecting sports-ball (Which is a Coco object). In order to do achieve that, I created a test set where I only labeled sports balls in objects and use this command to calculate several metrics:

`python val.py --weights yolov5x.pt --data dataset.yaml --img 640 --task test `

However, this yields low mAp scores since additional objects in the test dataset are also expected to be properly labeled but I only labeled sports balls. How can I calculate evaluation metrics for sports balls only?

**Comment by bryanbocao** [comment]
```
<data>.yaml (coco in this example)
    nc=1
    names=['sports_balls']
<model>.yaml
    nc=1
```
I would look at ```precision``` since it's single object.
```mAP``` is the (1) mean of all classes and (2) average of precision among all thresholds.

**Comment by glenn-jocher** [comment]
@bryanbocao to calculate evaluation metrics for sports balls only, you can modify the yaml files to set the number of classes to 1 and specify the class name as 'sports_balls'. Then, when evaluating the model, you can focus on the precision metric since you are dealing with a single object class. Keep in mind that mAP takes into account the mean precision across all classes as well as the average precision among all thresholds.

If you have any other questions or need further assistance, feel free to ask!
